mixins-before-declarations Errors

Hello,
When i used you sass lint, i found this error which i found little bit confused, normally it's shouldn't cause this error.
Could you please explain to me the problem ?
Thanks

```SASS
.area-current {
  $active: #00acac;
  $background: #2d353c;

  @include area-menu($active, $background);
}

.area-new {
  $active: #348fe2;
  $background: #2e475d;

  @include area-menu($active, $background);
}

.area-disappeared {
  $active: #727cb6;
  $background: #424a61;

  @include area-menu($active, $background);
}
```

```TXT
botify-report/sass/_themes.scss
   96:3  error  Mixins should come before declarations  mixins-before-declarations
  103:3  error  Mixins should come before declarations  mixins-before-declarations
  110:3  error  Mixins should come before declarations  mixins-before-declarations
```

--- a/lib/rules/mixins-before-declarations.js
+++ b/lib/rules/mixins-before-declarations.js
@@ -18,13 +18,22 @@ module.exports = {
       var depth = 0,
           declarationCount = [depth];
 
-      parent.traverse( function (item) {
+      parent.forEach( function (item) {
         if (item.type === 'ruleset') {
           depth++;
           declarationCount[depth] = 0;
         }
         else if (item.type === 'declaration') {
-          declarationCount[depth]++;
+          if (item.first().is('property')) {
+
+            var prop = item.first();
+
+            if(prop.first().is('ident')) {
+
+              declarationCount[depth]++;
+
+            }
+          }
         }
         else if (item.type === 'include') {
           item.forEach('simpleSelector', function (name) {

--- a/lib/rules/mixins-before-declarations.js
+++ b/lib/rules/mixins-before-declarations.js
@@ -28,7 +28,7 @@ module.exports = {
 
             var prop = item.first();
 
-            if(prop.first().is('ident')) {
+            if (prop.first().is('ident')) {
 
               declarationCount[depth]++;
 

--- a/tests/rules/mixins-before-declarations.js
+++ b/tests/rules/mixins-before-declarations.js
@@ -20,7 +20,7 @@ describe('mixins before declarations', function () {
   //////////////////////////////
   // Mixins Before Declarations - overwrite
   //////////////////////////////
-  it('[excludes]', function (done) {
+  it('[excludes: all]', function (done) {
     lint.test(file, {
       'mixins-before-declarations': [
         1,
@@ -35,6 +35,28 @@ describe('mixins before declarations', function () {
       ]
     }, function (data) {
       lint.assert.equal(0, data.warningCount);
+      done();
+    });
+  });
+
+  //////////////////////////////
+  // Mixins Before Declarations - exclude certain
+  // mixin test-again is not excluded
+  //////////////////////////////
+  it('[excludes: limited]', function (done) {
+    lint.test(file, {
+      'mixins-before-declarations': [
+        1,
+        {
+          'exclude': [
+            'waldo',
+            'mq',
+            'breakpoint'
+          ]
+        }
+      ]
+    }, function (data) {
+      lint.assert.equal(1, data.warningCount);
       done();
     });
   });

--- a/tests/sass/mixins-before-declarations.scss
+++ b/tests/sass/mixins-before-declarations.scss
@@ -1,3 +1,6 @@
+
+@include waldo;
+
 .foo {
   @include waldo;
   content: 'baz';
@@ -46,4 +49,29 @@
       @include test-again;
     }
   }
+}
+
+// added from issue #230 https://github.com/sasstools/sass-lint/issues/230
+
+.area-current {
+  $active: #00acac;
+  $background: #2d353c;
+
+  @include area-menu($active, $background);
+}
+
+// added from issue #227 https://github.com/sasstools/sass-lint/issues/227
+
+@include hello;
+
+@media (min-width: 50em) {
+  @include hello;
+}
+
+@supports (display: flex) {
+  @include hello;
+}
+
+@at-root {
+  @include hello;
 }

